frontend/src/lib/otel/shared.ts

```diff
@@ -86,8 +83,32 @@ export const traceHeaders = (span: Span, type: 'request' | 'response', headers:
   });
 };
 
+function getUser(event: RequestEvent | NavigationEvent | Event): LexAuthUser | null {
+  if (isBrowserEvent(event)) {
+    try {
+      const data = get(page).data;
+      return data.user as LexAuthUser;
+    } catch {
+      return null;
+    }
+  } else if (isRequestEvent(event)) {
+    return event.locals.getUser();
+  } else {
+    try {
+      const data = get(page).data;
+      return data.user as LexAuthUser;
+    } catch {
+      return null;
+    }
+  }
+}
+
 export const traceEventAttributes = (span: Span, event: RequestEvent | NavigationEvent | Event): void => {
   if (isBrowserEvent(event)) {
```

Comment:
Yes, you're right. I originally meant to have getUser called inside the else clause, but the way I wrote it, it can now be moved outside the if and we'll be fine.